### ISSUE TITLE
Commits without changes cannot be selected

### ISSUE THREAD

**Issue by ueffel** [ForkIssues/TrackerWin#1726]
Affected version: 1.80

Selecting a commit that has no file changes has no effect to the revision details view in the bottom.
It stays empty if no previous commit was selected like this:
![image](https://user-images.githubusercontent.com/2247384/207089226-4ed421af-3509-4d79-97b1-d0e9f4bb1ccb.png)
or it just keeps showing the previously selected commit with file changes.
![image](https://user-images.githubusercontent.com/2247384/207091770-1cdd8a50-3763-4f85-87cb-f685516d1f30.png)

1.79.2 showed these just fine.

Simple way to reproduce:

```sh
git init
echo foo > bar.txt
git add bar.txt
git commit -m initial
git commit --allow-empty -m initial++
```

In my case these empty commits are sometimes created by the build system, sometimes they're merge commits where no actual changes are left after conflict resolving.

**Comment by DanPristupov** [comment]
Thank you for the bug report

**Comment by DanPristupov** [comment]
Is that a critical problem for you?

**Comment by ueffel** [comment]
It's not critical, it's an inconvenience. A fix within 2 weeks would be fine.

**Comment by LondonRain** [comment]
We found the same bug and would be happy when it will be fixed. Thanks for reporting and taking care of it.

**Comment by lvernaillen** [comment]
Same here.
If it stays empty, because no commit was selected before, it's inconvenient but not critical.
However it's very confusing to see the changes of a previously selected commit in the change list of the currently selected empty commit.

For example: sometimes merge commit of a release branch merged into the main branch is empty, because the changes existed already on both branches. However with this bug, the commit that should have been empty now shows "random" changes as if someting on release branch was still missing on the main branch, while in reality it's not. "Random" in the sense that is shows the changes of whatever commit that was selected previously. So you can't trust the changes panel anymore when verifying the commits in the list view.

Thanks in advance for picking this up!

**Comment by DanPristupov** [comment]
The problem is fixed in Fork 1.81

**Comment by ueffel** [comment]
Confirmed fixed.
